articles/js-and-web-apis/index.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/js-and-web-apis/
+title: How Javascript Uses Web Browser Apis to Achieve Asynchronisity
+description: 
+author: adrian-murage
+date: 
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/js-and-web-apis/hero.jpg
+    alt: JS and Web APIs
+---
+### Introduction
+Pure JavaScript has a single thread of execution. Known as the [**main thread**](https://developer.mozilla.org/en-US/docs/Glossary/main_thread) in the browser context. "This means that long-running JavaScript functions can block the thread, leading to an unresponsive page and a bad user experience."[(MDN docs)](https://developer.mozilla.org/en-US/docs/Glossary/main_thread)
+
+Examples of long-running JavaScript functions are those that:
+- Fetch data from an API.
+- Wait for some time to elapse before executing.
+
+But, the above-mentioned functionality is commonly used in today's web applications. In this guide, you will learn how this functionality is achieved Asynchronously with Web Browser APIs.
```

Comment:
"The above-mentioned" remove the but
"asynchronously with web browser APIs" <- no capitalization expect for the acronym